### PR TITLE
refactor: user component and setup automatic docs

### PR DESCRIPTION
**Pull request by snqb**
- set up automatic docs generation from typescript
- refactor `User` component
- move out `User` out of badges
- change default SB view from canvas to docs
- add some more stories for User/Alternative header
- rename `User.stories.tsx` to `stories.tsx` as they're in a folder

image

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 module.exports = {
   "stories": [
-    "../src/**/*.stories.mdx",
```
Comment (snqb) [inline .storybook/main.js:3]: we don't use `.mdx` format

File: .storybook/main.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-    "../src/**/*.stories.@(js|jsx|ts|tsx)"
+    "../src/**/*/stories.@(ts|tsx)",
```
Comment (snqb) [inline .storybook/main.js:3]: now we can just name stories as `stories.tsx`, the prefix is redundant imo, as the file is in the directory anyways

File: src/components/alternative-header/stories.tsx [hunk 3]
```diff
@@ -52,27 +56,8 @@ const Template: Story<AlternativeHeaderProps> = (props) => {
...
-Default.args = {
-  title: 'Cluster settings',
-};
-
-Default.argTypes = {
-  title: {
-    summary: 'Title',
-    control: {
-      type: 'text',
-    },
```
Comment (snqb) [inline src/components/alternative-header/stories.tsx:64]: these now come from typescript directly

File: src/components/avatar/avatar.styles.ts [hunk 4]
```diff
@@ -1,7 +0,0 @@
-import { SxStyleProp } from 'rebass';
-
-export default {
```
Comment (snqb) [inline src/components/avatar/avatar.styles.ts:3]: just inlined it into a single place it was used

**Review by mstrluke — APPROVED** [review]
Good!

**Pull request merged**